pkg/cmd/route-controller-manager/cmd.go

```diff
@@ -0,0 +1,136 @@
+	options := &RouteControllerManager{Output: out}
+
+	cmd := &cobra.Command{
+		Use:   name,
+		Short: "Start the additional Ingress and Route controllers",
+		Long:  longDescription,
+		Run: func(c *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Validate())
+
+			serviceability.StartProfiler()
+
+			if err := options.StartControllerManager(stopCh); err != nil {
+				if kerrors.IsInvalid(err) {
+					if details := err.(*kerrors.StatusError).ErrStatus.Details; details != nil {
+						fmt.Fprintf(errout, "Invalid %s %s\n", details.Kind, details.Name)
+						for _, cause := range details.Causes {
+							fmt.Fprintf(errout, "  %s: %s\n", cause.Field, cause.Message)
+						}
+						os.Exit(255)
+					}
+				}
+				klog.Fatal(fmt.Sprintf("fatal error occurred: %v", err))
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+	// This command only supports reading from config
+	flags.StringVar(&options.ConfigFilePath, "config", options.ConfigFilePath, "Location of the master configuration file to run from.")
+	cmd.MarkFlagFilename("config", "yaml", "yml")
+	cmd.MarkFlagRequired("config")
+
+	return cmd
+}
+
+func (o *RouteControllerManager) Validate() error {
+	if len(o.ConfigFilePath) == 0 {
+		return errors.New("--config is required for this command")
+	}
+
+	return nil
+}
+
+// StartControllerManager calls RunControllerManager and then waits forever
+func (o *RouteControllerManager) StartControllerManager(stopCh <-chan struct{}) error {
+	if err := o.RunControllerManager(); err != nil {
+		return err
+	}
+
+	go daemon.SdNotify(false, "READY=1")
+	select {
+	case <-stopCh:
+		klog.Infof("Route Controller Manager received stop signal. exiting.")
+		return nil
+	}
+}
+
+// RunControllerManager takes the options and starts the controllers
+func (o *RouteControllerManager) RunControllerManager() error {
+	// try to decode into our new types first.  right now there is no validation, no file path resolution.  this unsticks the operator to start.
+	// TODO add those things
+	configContent, err := ioutil.ReadFile(o.ConfigFilePath)
+	if err != nil {
+		return err
+	}
+	scheme := runtime.NewScheme()
+	utilruntime.Must(openshiftcontrolplanev1.Install(scheme))
+	codecs := serializer.NewCodecFactory(scheme)
+	obj, err := runtime.Decode(codecs.UniversalDecoder(openshiftcontrolplanev1.GroupVersion, configv1.GroupVersion), configContent)
+	if err != nil {
+		return err
+	}
+
+	// Resolve relative to CWD
+	absoluteConfigFile, err := api.MakeAbs(o.ConfigFilePath, "")
+	if err != nil {
+		return err
+	}
+	configFileLocation := path.Dir(absoluteConfigFile)
+
+	config := obj.(*openshiftcontrolplanev1.OpenShiftControllerManagerConfig)
+	/// this isn't allowed to be nil when by itself.
+	// TODO remove this when the old path is gone.
+	if config.ServingInfo == nil {
+		config.ServingInfo = &configv1.HTTPServingInfo{}
+	}
+	// TODO: resolve split from OCM
+	config.ServingInfo.BindAddress = "0.0.0.0:9443"
```

Comment:
Where does `9443` port number come from?

Reply:
I have been using this for testing, eg. running this in the same pod as openshift-controller-manager, but since we will be moving this to a separate pod and ns I can put back the default 8443